apps/studio/src/pages/sites/[siteId]/settings.tsx

```diff
@@ -1,8 +1,175 @@
+import { useEffect, useState } from "react"
+import { useParams } from "next/navigation"
+import { useRouter } from "next/router"
+import {
+  Button,
+  Center,
+  HStack,
+  Text,
+  useDisclosure,
+  VStack,
+} from "@chakra-ui/react"
+import {
+  Infobox,
+  Input,
+  Toggle,
+  useToast,
+} from "@opengovsg/design-system-react"
+
+import { UnsavedSettingModal } from "~/features/editing-experience/components/UnsavedSettingModal"
 import { type NextPageWithLayout } from "~/lib/types"
 import { AdminCmsSidebarLayout } from "~/templates/layouts/AdminCmsSidebarLayout"
+import { trpc } from "~/utils/trpc"
 
 const SiteSettingsPage: NextPageWithLayout = () => {
-  return <div>Settings page</div>
+  const toast = useToast()
+  const router = useRouter()
+  const params = useParams<{ siteId: string }>()
```

Comment:
nit: added a `useQueryParse` hook somewhere here to parse queries or throw.
Can probably use it

```ts
// outside component
const siteSettingsSchema = z.object({
  siteId: z.coerce.number(),
})

// usage
const { siteId } = useQueryParse(siteSettingsSchema)
```